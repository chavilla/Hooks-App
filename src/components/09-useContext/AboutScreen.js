import React, { useContext } from 'react';
import { UserContext } from './UserContext';

const AboutScreen = () => {

    const {user, setUser} = useContext(UserContext);

    const handleLogout=()=>{
        setUser({
        });
    }

    return ( 
        <div className='container mt-5'>
            <pre>
            {JSON.stringify(user,null,3)}
            </pre>

            <button className='btn btn-warning'
            onClick={handleLogout}
            >
                Logout
            </button>
        </div>
     );
}
 
export default AboutScreen;