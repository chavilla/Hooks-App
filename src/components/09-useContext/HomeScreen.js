import React, { useContext } from 'react';
import { UserContext } from './UserContext';

const HomeScreen = () => {

    const user = useContext(UserContext);

    return ( 
        <div className='container my-5'>
            <pre>
            homescreen {JSON.stringify(user,null,3)} 
            </pre>
        </div>
     );
}
 
export default HomeScreen;