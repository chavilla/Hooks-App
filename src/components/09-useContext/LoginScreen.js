import React, { useContext } from 'react';
import { UserContext } from './UserContext';

const LoginScreen = () => {

    //Ontener referencia al context
    const { user,setUser} = useContext(UserContext);
    
    //objeto con id y name
    const newUser={
        id:3,
        name:'Luis',
        email: 'lualchavi@hotmail.com'
    }

    return ( 
        <div className='container'>
            <p className='d-block mt-5'>Login</p>
            <button type='button'
            className='btn btn-primary'
            onClick={()=>setUser(newUser)}
            >
                Agregar Usuario
            </button>
            { JSON.stringify(user) }
        </div>
     );
}
 
export default LoginScreen;