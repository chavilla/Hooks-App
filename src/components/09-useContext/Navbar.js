import React from "react";
import { Link, NavLink } from "react-router-dom";

const Navbar = () => {
  return (
    <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
      <Link to="/" className=''><p className='text-white'>UseContext</p></Link>

      <div className="collapse navbar-collapse ml-5" id="navbarNavAltMarkup">
        <div className="navbar-nav">
          <NavLink
            exact
            activeClassName="active"
            className="nav-link"
            to="/"
          >
            Home
          </NavLink>

          <NavLink
            exact
            activeClassName="active"
            className="nav-link"
            to="/login"
          >
            Login
          </NavLink>

          <NavLink
            exact
            activeClassName="active"
            className="nav-link"
            to="/about"
          >
            About
          </NavLink>
        </div>
      </div>
    </nav>
  );
};

export default Navbar;
