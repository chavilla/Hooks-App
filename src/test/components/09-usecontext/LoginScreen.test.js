import React from 'react';
import { mount } from 'enzyme';
import { UserContext } from '../../../components/09-useContext/UserContext';
import LoginScreen from '../../../components/09-useContext/LoginScreen';

describe('<LoginScreen />', () => {

    const setUser=jest.fn()

    const wrapper=mount(
    <UserContext.Provider value={{ setUser }}>
        <LoginScreen />
    </UserContext.Provider>
    );


    test('display the component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('execute setUser function', () => {
        
        wrapper.find('button').prop('onClick')();

        expect(setUser).toHaveBeenCalledWith( {id:3,
            name:'Luis',
            email: 'lualchavi@hotmail.com'})
    })
    
    

})