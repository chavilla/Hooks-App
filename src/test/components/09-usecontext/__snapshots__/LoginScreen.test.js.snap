// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginScreen /> display the component 1`] = `
<div
  className="container"
>
  <p
    className="d-block mt-5"
  >
    Login
  </p>
  <button
    className="btn btn-primary"
    onClick={[Function]}
    type="button"
  >
    Agregar Usuario
  </button>
</div>
`;
